courses/views_test.py

```diff
@@ -334,18 +356,22 @@ def test_program_view(client, user, home_page, is_enrolled, has_product, is_anon
     assert resp.context["product_id"] == product_id
     assert resp.context["enrolled"] == (is_enrolled and not is_anonymous)
 
-    has_button = False
+    # Anynoymous users don't see the enrolled/enroll-now button.
```

Comment:
I believe, Anynoymous users can see the enrolled/enroll-now button but on click over it they will be redirected to `Login` Page.

Reply:
Anonymous users see a sign-in/create-account button, not enroll now button. Specifically, this check is for the button with `enroll-button` class which is different from the sign-in/create-account button class.

Reply:
I belive, there should be enroll now button for Anonymous users too, Am I missing something over it ?

Reply:
Ok. seems `Enroll Now` should not be appeared over `HomePage` but it should be over `ProductDetail` page. A bug fix ?

Reply:
Fix is included in the PR.